R/plot_MutationTimeR.R

```diff
@@ -3,40 +3,39 @@
 #' @description This function plots the timing information from MutationTimeR for SSMs and CNAs.
 #'
 #' @param this_sample_metadata Metadata with one row containing your sample details
-#' @param projection Genome build projection (e.g., "hg38" or "grch37").
+#' @param timed_cna Timed CNAs from `GAMBLR.results::get_timed_mutations`
+#' @param timed_ssm Timed SSMs from `GAMBLR.results::get_timed_mutations`
 #' @param verbose Set to TRUE for a chattier experience. Default is FALSE.
 #' @import ggplot2 dplyr ggpubr
 #' @export
-#' @examples 
-#' 
-#' my_meta = suppressMessages(get_gambl_metadata()) %>% 
+#' @examples
+#'
+#' my_meta = suppressMessages(get_gambl_metadata()) %>%
 #'   dplyr::filter(sample_id=="01-20985T",
 #'                seq_type=="genome")
 #' timed = GAMBLR.results::get_timed_mutations(my_meta,"hg38")
 #' print(timed$SSM)
-#' 
+#'
 #' print(timed$CNA)
-#' 
+#'
 #' all_plots = plot_MutationTimeR(my_meta,timed$CNA,timed$SSM)
-#' 
+#'
 #' all_plots$full
-#' 
+#'
 #' all_plots$minimal
-plot_MutationTimeR <- function(this_sample_metadata, 
-                              timed_cna, 
-                              timed_ssm, 
-                              genome_build, 
+plot_MutationTimeR <- function(this_sample_metadata,
+                              timed_cna,
+                              timed_ssm,
                               verbose = FALSE) {
   if (nrow(this_sample_metadata) > 1) {
     stop("this_sample_metadata must contain exactly one row")
   }
-  if(missing(genome_build)){
-    genome_build = get_genome_build(timed_cna)
-  }
+
```

Comment:
This is more of an FYI/optional suggestion as I haven't started doing this systematically but it another check we could add is to assess whether the inputs are of the expected class. Something like these examples:
```
if(!"seg_data" %in% class(timed_cna)){
  stop("timed_cna must be a seg_data object")
}
if(!"maf_data" %in% class(timed_ssm)){
  stop("timed_ssm must be a maf_data object")
}
if(!"some_expected_column_from_mutationTimer" %in% colnames(timed_ssm){
stop("timed_ssm does not appear to be an output from MutationTimer")
}
```
you can imagine this could become quite lengthy and some of these could be supported by as-yet nonexistent helper functions